app/views/projects/_status_chooser.html.erb

```diff
@@ -1,17 +1,15 @@
 <%
-   status_symbol = (criterion.to_s + '_status').to_sym
-   justification_symbol = (criterion.to_s + '_justification').to_sym
-   project_criterion_status = project[status_symbol]
-   project_criterion_justification = project[justification_symbol] %>
+ status_symbol = (criterion.to_s + '_status').to_sym
+ project_criterion_status = project[status_symbol]
+ criterion_result = project.get_criterion_result(criterion)
 
-<% cache [criterion.to_s, criteria_level, is_disabled, locale,
-          project_criterion_status, project_criterion_justification ] do
-     criterion_result = project.get_criterion_result(criterion)
-     passing_class = ''
-      if criterion_result == :criterion_passing && is_disabled
-       passing_class = ' criterion-passing'
-     end
-  %>
+ cache ['buttons', criterion.to_s, criteria_level, is_disabled, locale,
+        project_criterion_status ] do
+   passing_class = ''
+   if criterion_result == :criterion_passing && is_disabled
+     passing_class = ' criterion-passing'
+   end
+%>
```

Comment:
I guess it makes it a little more obvious that the final state has 2 options. Okay, I'll try it.

I would do that often in a functional language, but Ruby is definitely not a functional language.